pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -233,6 +236,19 @@ func init() {
 
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
 func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {
```

Comment:
Is this method getting invoked in Kubelet? If no, then we should fail the initialization if vs.informerFactory is not set.

Can you confirm if Initialize is invoked after SetInformers?

You are also introducing a vsphere-cloud-provider service account, wouldn't this break Kubernetes upgrade? How do you plan to address upgrade?

Reply:
I think we will have to install corresponding rbac policies as defaults.

Reply:
This does not get called from kubelet. Also `Initialize` is called after `SetInformers`, so I think we should be safe. Having said that, I think the code should work if it can't find configmap as well and revert to old behaviour when configmap isn't found.